### PR TITLE
Feature/consultation pages

### LINKED ISSUE #48: Consultation list and details page with network layer
similar to group pages

### LINKED ISSUE #49: Rework consultation create/edit form with RHF
ReactHookForm
and make it submittable to the backend

or no RHF, but make it work with backend

### PR DESCRIPTION
**Pull request by DannySS123**
Should be enough for now.

- Closes #48 
- Closes #49

### CODE REVIEW

File: src/pages/consultations/ConsultationDetailsPage.tsx [hunk 1]
```diff
@@ -49,20 +74,67 @@ export const ConsultationDetailsPage = () => {
...
+                    <Button
+                      onClick={() => {
+                        joinConsultation(+consultationId!!)
+                      }}
+                      colorScheme="brand"
+                    >
+                      Megyek
+                    </Button>
+                  </>
+                )}
```
Comment (Tschonti) [inline src/pages/consultations/ConsultationDetailsPage.tsx:136]: owner shouldn't be able to join, even if they aren't presenting. (Or maybe they should, but then change the permissions on the backend xd)

**Review by Tschonti — CHANGES_REQUESTED** [review]
Úristen very big 😮
Thanks for all the work put into this! I left a lot of comments, but that's just because you did so much xd (but also left a few things unfinished). Also, these are the most important pages of the site, so we need to get the UX just right. 
One more thing: You can add the same presenter twice (though that modal seems to be unfinished)

**Review by Tschonti — CHANGES_REQUESTED** [review]
Very nice, only one problem remains: You can add the same user multiple times as a presenter. The backend ignores the duplicate IDs, so it's not a huge issue, but should still be fixed.

**Review by Tschonti — APPROVED** [review]
🥇

File: src/pages/consultations/components/UserRating.tsx [hunk 2]
```diff
@@ -13,53 +17,109 @@ import {
...
+  )
+
   const [open, setOpen] = useState(false)
-  const [value, setValue] = useState<number>(user?.rating?.value ?? 5)
-  const [text, setText] = useState<string>(user?.rating?.text ?? '')
+  const [value, setValue] = useState<number>(user.rating?.value ?? 5)
+  const [text, setText] = useState<string>(user.rating?.text ?? '')
+  const [anonymous, setAnonymous] = useState<boolean>(user.rating?.anonymous ?? false)
+
+  if (consultationId === undefined || isNaN(+consultationId)) {
```
Comment (berenteb) [inline src/pages/consultations/components/UserRating.tsx:71]: id validation outsource can be used here for example, the function mentioned above

File: src/pages/consultations/components/UserRating.tsx [hunk 3]
```diff
@@ -13,53 +17,109 @@ import {
...
+      toast(generateToastParams(e))
+    }
+  )
+
   const [open, setOpen] = useState(false)
-  const [value, setValue] = useState<number>(user?.rating?.value ?? 5)
-  const [text, setText] = useState<string>(user?.rating?.text ?? '')
+  const [value, setValue] = useState<number>(user.rating?.value ?? 5)
+  const [text, setText] = useState<string>(user.rating?.text ?? '')
+  const [anonymous, setAnonymous] = useState<boolean>(user.rating?.anonymous ?? false)
```
Comment (berenteb) [inline src/pages/consultations/components/UserRating.tsx:69]: use i.e. !!user.rating?.anonymous

File: src/pages/error/ErrorPage.tsx [hunk 4]
```diff
@@ -19,21 +19,22 @@ export const ErrorPage = ({ title, messages, backPath, status }: ErrorPageState)
...
   } = (state as ErrorPageState) || {
     title: 'This is a blank error page',
     messages: [], //['You might have found yourself here by reloading the error page or copying a link from somewhere else.'],
     backPath: '/'
   }
 
   useEffect(() => {
-    if (status === 401) {
+    if (status === 401 || s === 401) {
       toast({ title: '401 Nem vagy bejelentkezve', status: 'error' })
```
Comment (berenteb) [inline src/pages/error/ErrorPage.tsx:32]: Omit 401 here imo, the user does not care about the status code

File: src/pages/groups/GroupDetailsPage.tsx [hunk 5]
```diff
@@ -10,60 +10,55 @@ import { UserList } from './components/UserList'
 
 export const GroupDetailsPage = () => {
   const { groupId } = useParams()
-  if (groupId === undefined) {
-    return <ErrorPage />
+  const { isLoading, data: group, error, refetch } = useFecthGroupDetailsQuery(+groupId!!)
+
+  if (groupId === undefined || isNaN(+groupId)) {
```
Comment (berenteb) [inline src/pages/groups/GroupDetailsPage.tsx:15]: isIdValid func use case

**Review by berenteb — CHANGES_REQUESTED** [review]
Nice work all together, no red flags coded 🫡
Some minor code quality suggestions in there. Some thoughts overall:
- Use datefns id you can, it will make your life easier
- Create enums or sth for paths, and idea: create a function for joining paths with '/'. Less developer errors and more consistency
- At some point outsourcing strings will be straightforward for the future
- We might need to configure prettier and eslint, also enabling some kind of pipeline for them could be useful.

File: src/api/hooks/consultationQueryHooks.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  })
+}
+
+export const useFetchConsultationbDetailsQuery = (consultationId: number) => {
+  return useQuery<ConsultationDetails, KonziError>(
+    ['fetchConsultationDetails', consultationId],
+    async () => (await axios.get(`/consultations/${consultationId}`)).data,
+    {
+      retry: false,
+      enabled: consultationId > 0 && !isNaN(consultationId)
```
[the remark this answers is not in the document]
Comment (Tschonti) [reply src/api/hooks/consultationQueryHooks.ts:19]: https://github.com/kir-dev/konzisite-frontend/issues/80

File: src/api/hooks/groupMutationHooks.ts [hunk 7]
```diff
@@ -10,6 +11,14 @@ interface GroupMutationParams {
   userId: number
 }
 
+export const useFecthGroupListMutation = (onError: (e: KonziError) => void) => {
+  return useMutation<GroupPreview[], KonziError, string>(
+    'fetchGroupsMuatation',
+    async (search: string) => (await axios.get(`/groups?search=${search}`)).data,
```
[the remark this answers is not in the document]
Comment (Tschonti) [reply src/api/hooks/groupMutationHooks.ts:17]: https://github.com/kir-dev/konzisite-frontend/issues/81

File: src/api/hooks/subjectHooks.ts [hunk 8]
```diff
@@ -9,6 +9,14 @@ export const useFetchSubjectsQuery = () => {
   return useQuery<SubjectModel[], KonziError>('fetchSubject', async () => (await axios.get('/subjects')).data, { retry: false })
 }
 
+export const useFecthSubjectListMutation = (onError: (e: KonziError) => void) => {
+  return useMutation<SubjectModel[], KonziError, string>(
+    'fetchSubjectsMuatation',
+    async (search: string) => (await axios.get(`/subjects?search=${search}`)).data,
```
[the remark this answers is not in the document]
Comment (Tschonti) [reply src/api/hooks/subjectHooks.ts:15]: https://github.com/kir-dev/konzisite-frontend/issues/81

File: src/api/hooks/userMutationHooks.ts [hunk 9]
```diff
@@ -0,0 +1,12 @@
+import axios from 'axios'
+import { useMutation } from 'react-query'
+import { UserPreview } from '../../pages/user/types/UserPreview'
+import { KonziError } from '../model/error.model'
+
+export const useFecthUserListMutation = (onError: (e: KonziError) => void) => {
+  return useMutation<UserPreview[], KonziError, string>(
+    'fetchUsersMuatation',
+    async (search: string) => (await axios.get(`/users?search=${search}`)).data,
```
[the remark this answers is not in the document]
Comment (Tschonti) [reply src/api/hooks/userMutationHooks.ts:9]: https://github.com/kir-dev/konzisite-frontend/issues/81

File: src/pages/consultations/components/UserRating.tsx (hunk 2 above)
Comment (Tschonti) [reply src/pages/consultations/components/UserRating.tsx:71]: https://github.com/kir-dev/konzisite-frontend/issues/80

File: src/pages/groups/GroupDetailsPage.tsx (hunk 5 above)
Comment (Tschonti) [reply src/pages/groups/GroupDetailsPage.tsx:15]: https://github.com/kir-dev/konzisite-frontend/issues/80

File: src/pages/error/ErrorPage.tsx (hunk 4 above)
Comment (Tschonti) [reply src/pages/error/ErrorPage.tsx:32]: https://github.com/kir-dev/konzisite-frontend/issues/82

File: src/pages/consultations/ConsultationDetailsPage.tsx [hunk 10]
```diff
@@ -1,85 +1,166 @@
...
+        </Box>
       )}
+      <Heading size="lg" mb={2}>
+        Konzitartók ({consultation.presentations.length})
+      </Heading>
+      <UserList
+        columns={1}
+        users={consultation.presentations}
+        isParticipant={isParticipant}
+        showRatingButton={new Date(consultation.startDate).getTime() < new Date().getTime()}
```
[the remark this answers is not in the document]
Comment (Tschonti) [reply src/pages/consultations/ConsultationDetailsPage.tsx:156]: startDate is actually a DateTime object, so the epoch time is important here. We only show the button after the konzi has started

File: src/pages/consultations/EditConsultationPage.tsx [hunk 11]
```diff
@@ -1,42 +1,102 @@
...
+        }
+      }
+    )
+  }
+
   const [loading, setLoading] = useState(!newConsultation)
   const [name, setName] = useState('')
   const [location, setLocation] = useState('')
-  const [description, setDescription] = useState<string | undefined>('')
+  const [description, setDescription] = useState<string>('')
```
[the remark this answers is not in the document]
Comment (Tschonti) [reply src/pages/consultations/EditConsultationPage.tsx:85]: this part has been removed in #78

File: src/pages/consultations/EditConsultationPage.tsx [hunk 12]
```diff
@@ -1,42 +1,102 @@
...
   const [startDate, setStartDate] = useState(new Date())
   const [endDate, setEndDate] = useState(new Date())
   const [subject, setSubject] = useState<SubjectModel | undefined>(undefined)
```
[the remark this answers is not in the document]
Comment (Tschonti) [reply src/pages/consultations/EditConsultationPage.tsx:88]: this part has been removed in #78

File: src/pages/consultations/EditConsultationPage.tsx [hunk 13]
```diff
@@ -1,42 +1,102 @@
...
+  let locationError = location === ''
   let startDateError = newConsultation
     ? startDate.getTime() <= new Date().getTime()
-    : startDate.getTime() < Math.min(consultation?.startDate?.getTime() ?? 0, new Date().getTime())
+    : startDate.getTime() < Math.min(consultation ? new Date(consultation.startDate).getTime() : 0, new Date().getTime())
   let endDateError = endDate.getTime() <= startDate.getTime()
-  let subjectError = subject == undefined
-  let presentationsError = presentations.length == 0
+  let subjectError = subject === undefined
+  let presentationsError = presentations.length === 0
```
[the remark this answers is not in the document]
Comment (Tschonti) [reply src/pages/consultations/EditConsultationPage.tsx:99]: this part has been removed in #78

File: src/pages/consultations/ConsultationDetailsPage.tsx [hunk 14]
```diff
@@ -1,85 +1,166 @@
...
+
+  if (error) {
+    return <ErrorPage backPath={'/'} status={error.statusCode} title={error.message} />
+  }
+
+  if (isLoading || loggedInUserLoading) {
+    return <LoadingConsultation />
+  }
+
+  if (consultation === undefined) {
```
[the remark this answers is not in the document]
Comment (Tschonti) [reply src/pages/consultations/ConsultationDetailsPage.tsx:59]: #84

File: src/pages/consultations/EditConsultationPage.tsx [hunk 15]
```diff
@@ -50,88 +110,92 @@ export const EditConsultationPage = ({ newConsultation }: Props) => {
...
-                endDate={endDate}
-                setStartDate={setStartDate}
-                setEndDate={setEndDate}
-                startDateError={startDateError}
-                endDateError={endDateError}
+  if (loading || isLoading || loggedInUserLoading) {
+    return <LoadingEditConsultation />
+  }
+
+  if (loggedInUser === undefined) {
```
[the remark this answers is not in the document]
Comment (Tschonti) [reply src/pages/consultations/EditConsultationPage.tsx:134]: #84

File: src/pages/consultations/EditConsultationPage.tsx [hunk 16]
```diff
@@ -50,88 +110,92 @@ export const EditConsultationPage = ({ newConsultation }: Props) => {
...
+    return <ErrorPage status={401} />
+  }
+
+  return (
+    <>
+      {(consultation === undefined ||
```
[the remark this answers is not in the document]
Comment (Tschonti) [reply src/pages/consultations/EditConsultationPage.tsx:140]: #84

**Comment by Tschonti** [comment]
Gonna merge this, the remaining change requests have been converted to issues

**Pull request merged**